arwen/elrondapi/bigFloatOps.go

```diff
@@ -0,0 +1,893 @@
+	dest.Abs(op)
+}
+
+//export v1_4_bigFloatSign
+func v1_4_bigFloatSign(context unsafe.Pointer, opHandle int32) int32 {
+	managedType := arwen.GetManagedTypesContext(context)
+	metering := arwen.GetMeteringContext(context)
+	runtime := arwen.GetRuntimeContext(context)
+
+	gasToUse := metering.GasSchedule().BigFloatAPICost.BigFloatAbs
+	metering.UseGas(gasToUse)
+
+	op, err := managedType.GetBigFloat(opHandle)
+	if arwen.WithFault(err, context, runtime.BigFloatAPIErrorShouldFailExecution()) {
+		return -2
+	}
+	managedType.ConsumeGasForBigFloatCopy(op)
+	return int32(op.Sign())
+}
+
+//export v1_4_bigFloatSqrt
+func v1_4_bigFloatSqrt(context unsafe.Pointer, destinationHandle, opHandle int32) {
+	managedType := arwen.GetManagedTypesContext(context)
+	metering := arwen.GetMeteringContext(context)
+	runtime := arwen.GetRuntimeContext(context)
+
+	gasToUse := metering.GasSchedule().BigFloatAPICost.BigFloatSqrt
+	metering.UseGas(gasToUse)
+
+	dest := managedType.GetBigFloatOrCreate(destinationHandle)
+	op, err := managedType.GetBigFloat(opHandle)
+	if arwen.WithFault(err, context, runtime.BigFloatAPIErrorShouldFailExecution()) {
+		return
+	}
+	if op.Sign() < 0 {
+		_ = arwen.WithFault(arwen.ErrBadLowerBounds, context, runtime.BigFloatAPIErrorShouldFailExecution())
+		return
+	}
+	if oneIsInfinity(op) {
+		_ = arwen.WithFault(arwen.ErrInfinityFloatOperation, context, runtime.BigFloatAPIErrorShouldFailExecution())
+		return
+	}
+	managedType.ConsumeGasForBigFloatCopy(dest, op)
+	dest.Sqrt(op)
+}
+
+//export v1_4_bigFloatLog2
+func v1_4_bigFloatLog2(context unsafe.Pointer, opHandle int32) int32 {
+	managedType := arwen.GetManagedTypesContext(context)
+	metering := arwen.GetMeteringContext(context)
+	runtime := arwen.GetRuntimeContext(context)
+
+	gasToUse := metering.GasSchedule().BigFloatAPICost.BigFloatLog2
+	metering.UseGas(gasToUse)
+
+	op, err := managedType.GetBigFloat(opHandle)
+	if arwen.WithFault(err, context, runtime.BigFloatAPIErrorShouldFailExecution()) {
+		return -1
+	}
+	if op.Sign() < 0 {
+		_ = arwen.WithFault(arwen.ErrBadLowerBounds, context, runtime.BigFloatAPIErrorShouldFailExecution())
+		return -1
+	}
+	if oneIsInfinity(op) {
+		_ = arwen.WithFault(arwen.ErrInfinityFloatOperation, context, runtime.BigFloatAPIErrorShouldFailExecution())
+		return -1
+	}
+	bigIntOp := new(big.Int)
+	op.Int(bigIntOp)
+	managedType.ConsumeGasForBigFloatCopy(op)
+	managedType.ConsumeGasForBigIntCopy(bigIntOp)
+	return int32(bigIntOp.BitLen() - 1)
+}
+
+//export v1_4_bigFloatPow
+func v1_4_bigFloatPow(context unsafe.Pointer, destinationHandle, op1Handle, op2Handle int32) {
+	managedType := arwen.GetManagedTypesContext(context)
+	metering := arwen.GetMeteringContext(context)
+	runtime := arwen.GetRuntimeContext(context)
+
+	gasToUse := metering.GasSchedule().BigIntAPICost.BigIntPow
+	metering.UseGas(gasToUse)
+
+	dest := managedType.GetBigFloatOrCreate(destinationHandle)
+	op1, op2, err := managedType.GetTwoBigFloat(op1Handle, op2Handle)
```

Comment:
op2 can be a regulat small int, I don't think there will ever be enough gas for huge exponentiation. Big int probably overkill, definitely not float, since we are only doing repeated multiplications.